src/components/vm/hostdevs/hostDevCard.jsx

```diff
@@ -0,0 +1,216 @@
+        } else if (hostDev.type === "scsi_host") {
+            const protocol = hostDev.source.protocol;
+            const wwpn = hostDev.source.wwpn;
+
+            addOptionalToCell(cell, protocol, `${hostdevId}-protocol`, _("Protocol"));
+            addOptionalToCell(cell, wwpn, `${hostdevId}-wwpn`, _("WWPN"));
+        } else if (hostDev.type === "mdev") {
+            const uuid = hostDev.source.address.uuid;
+
+            addOptionalToCell(cell, uuid, `${hostdevId}-uuid`, _("UUID"));
+        } else if (hostDev.type === "storage") {
+            const block = hostDev.source.block;
+
+            addOptionalToCell(cell, block, `${hostdevId}-block`, _("Block"));
+        } else if (hostDev.type === "misc") {
+            const ch = hostDev.source.char;
+
+            addOptionalToCell(cell, ch, `${hostdevId}-char`, _("Char"));
+        } else if (hostDev.type === "net") {
+            const iface = hostDev.source.interface;
+
+            addOptionalToCell(cell, iface, `${hostdevId}-interface`, _("Interface"));
+        }
+
+        return cell;
+    }
+
+    const id = vmId(vm.name);
+
+    // Hostdev data mapping to rows
+    const detailMap = [
+        {
+            name: _("Type"), value: (hostdev, hostdevId) => {
+                return (
+                    <span id={`${id}-hostdev-${hostdevId}-type`}>
+                        {hostdev.type}
+                    </span>
+                );
+            }
+        },
+        {
+            name: _("Class"), value: (hostdev, hostdevId) => {
+                return (
+                    <span id={`${id}-hostdev-${hostdevId}-class`}>
+                        {getClass(hostdev, hostdevId)}
+                    </span>
+                );
+            }
+        },
+        {
+            name: _("Model"), value: (hostdev, hostdevId) => {
+                return (
+                    <div id={`${id}-hostdev-${hostdevId}-product`}>
+                        {getProduct(hostdev, hostdevId)}
+                    </div>
+                );
+            }
+        },
+        {
+            name: _("Vendor"), value: (hostdev, hostdevId) => {
+                return (
+                    <div id={`${id}-hostdev-${hostdevId}-vendor`}>
+                        {getVendor(hostdev, hostdevId)}
+                    </div>
+                );
+            }
+        },
+        {
+            name: _("Source"), value: (hostdev, hostdevId) => {
+                return (
+                    <div id={`${id}-hostdev-${hostdevId}-source`}>
+                        {getSource(hostdev, hostdevId)}
+                    </div>
+                );
+            }
+        },
+    ];
+
+    let hostdevId = 1;
+
+    const columnTitles = detailMap.map(target => target.name);
+    const rows = vm.hostDevices.sort((a, b) => a.type < b.type && -1).map(target => {
+        const columns = detailMap.map(d => {
+            let column = null;
+            if (typeof d.value === 'function') {
```

Comment:
There is nothing else than function here. You can remove the check and add const column = ...

Reply:
Fixed